src/coreclr/src/jit/codegenxarch.cpp

```diff
@@ -6821,13 +6821,20 @@ void CodeGen::genSSE2BitwiseOp(GenTree* treeNode)
         *bitMask = GetEmitter()->emitAnyConst(cnsAddr, cnsSize, cnsAlign);
     }
 
+    GenTree* op1 = treeNode->AsOp()->gtOp1;
+    assert(op1->isUsedFromReg());
+
+    if (compiler->canUseVexEncoding())
+    {
+        GetEmitter()->emitIns_R_R_C(ins, emitTypeSize(targetType), targetReg, genConsumeReg(op1), *bitMask, 0);
+        return;
+    }
+
     // We need an additional register for bitmask.
     regNumber tmpReg = treeNode->GetSingleTempReg();
```

Comment:
The register allocator should likely be updated to not request a temporary here.

I also wonder if we should update this to use `inst_RV_RV_TT` for the general codepath instead: https://github.com/dotnet/runtime/blob/master/src/coreclr/src/jit/codegenxarch.cpp#L902

Reply:
@tannergooding can take a look at my recent commit if it addresses your comment

Reply:
hm.. there is a problem with `inst_RV_RV_TT` - it always call `emitFltOrDblConst` even if we already have such const in the data section. The current code caches it.

Reply:
for instance, currently in master:
```csharp
float Egor(float x, float y) => 10 * x * 10;
```
```asm
; Method Z:Egor(float,float):float
G_M738_IG01:
       C5F877               vzeroupper 
						;; bbWeight=1    PerfScore 1.00

G_M738_IG02:
       C5FA59050D000000     vmulss   xmm0, xmm0, dword ptr [reloc @RWD00]
       C5FA590509000000     vmulss   xmm0, xmm0, dword ptr [reloc @RWD04]
						;; bbWeight=1    PerfScore 10.00

G_M738_IG03:
       C3                   ret      
						;; bbWeight=1    PerfScore 1.00
RWD00  dd	41200000h
RWD04  dd	41200000h
; Total bytes of code: 20
```

Reply:
I moved the cache logic for GT_NEG and Math(F).Abs masks to `inst_RV_RV_TT` and filed https://github.com/dotnet/runtime/issues/42178